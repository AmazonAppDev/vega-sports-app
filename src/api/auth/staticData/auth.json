{
  "auth": {
    "id": "nmbsd-234sdc-678-jklsdf",
    "name": "Freida",
    "surname": "Gomam",
    "email": "fgomam@fakemail.com",
    "address": "Nowhere 5 USA",
    "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "profiles": [
      {
        "id": "bada55-1dl00k-4tm33",
        "displayName": "Freida",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg"
      },
      {
        "id": "l00k-1nt0-s0uL",
        "displayName": "Jennifer",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg"
      }
    ]
  }
}
