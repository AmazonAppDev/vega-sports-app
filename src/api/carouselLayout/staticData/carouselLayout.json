{
  "carousellayout": [
    {
      "id": "e77effd2-60fd-4a75-9921-ee1596c4d248",
      "carouselType": "hero",
      "carouselTitle": null,
      "endpoint": "suggestedforyou"
    },
    {
      "id": "fac0db1d-60ee-4ff9-9c3f-fc1ca24fa6be",
      "carouselType": "square",
      "carouselTitle": "Live TV: fresh news from events",
      "endpoint": "livestreams"
    },
    {
      "id": "cf5bafe0-023b-4ff9-9c3f-fc1ca24fa6be",
      "carouselType": "card",
      "carouselTitle": "Sport Teams Hall of Fame",
      "endpoint": "teams"
    },
    {
      "id": "068f3841-023b-4de8-85d1-f410d1407528",
      "carouselType": "square",
      "carouselTitle": "Documentaries on Sport Stars",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc13a87-af56-42ef-a66e-b882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f1389-845c-476d-820b-1aa388e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d21039f-3ba5-491b-b481-324ceb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "fac0db1d-60ee-4ff9-9c3f-fc15ca24fa6be",
      "carouselType": "square",
      "carouselTitle": "Live TV: fresh news from events",
      "endpoint": "livestreams"
    },
    {
      "id": "cf5bafe0-023b-4ff9-9c3f-6fc1ca24fa6be",
      "carouselType": "card",
      "carouselTitle": "Sport Teams Hall of Fame",
      "endpoint": "teams"
    },
    {
      "id": "06844f3841-023b-4de8-85d1-7f410d14aa07528",
      "carouselType": "square",
      "carouselTitle": "Documentaries on Sport Stars",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc1113a87-af56-42ef-a66e8-b8jj8a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f11389-845c-476d-8290b-1aajj88e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d221039f-3ba5-491b-b4081-32cejjb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc132a87-af56-42ef-a66e-bgh882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c9177f14389-845c-476d-820b-s",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "9fc166113a87-af56-42ef-a66e8-b8jj8a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f5511389-845c-476d-8290b-1aajj88e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d22441039f-3ba5-491b-b4081-32cejjb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc13233a87-af56-42ef-a66e-bgh882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f1422389-845c-476d-820b-s",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    }
  ]
}
