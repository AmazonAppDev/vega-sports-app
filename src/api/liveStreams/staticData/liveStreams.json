{
  "livestreams": [
    {
      "id": "50128bae-e954-4233-8e15-cd5867a31370",
      "stream_date": "2022-12-17T04:52:43Z",
      "title": "Live from Century's Football Match",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/11.jpg",
      "sport_type": "surfing"
    },
    {
      "id": "d9f25760-7263-426a-b4d7-a25e14a26f03",
      "stream_date": "2022-06-09T11:58:23Z",
      "title": "Formula 1 Discussion Panel",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/12.jpg",
      "sport_type": "soccer"
    },
    {
      "id": "73780c3d-2f09-4ea0-b9c0-50bdf2a9ba0a",
      "stream_date": "2022-10-07T23:10:08Z",
      "title": "Inside the Championship Studio",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/13.jpg",
      "sport_type": "padel"
    },
    {
      "id": "4448fb6d-f55c-4975-8b0c-14d5356d807b",
      "stream_date": "2022-01-25T06:12:26Z",
      "title": "Post-Match Analysis Desk",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/14.jpg",
      "sport_type": "surfing"
    },
    {
      "id": "997afe05-d0fa-44c2-b62e-555062807623",
      "stream_date": "2022-08-20T07:38:40Z",
      "title": "Voices from the Sidelines",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/15.jpg",
      "sport_type": "surfing"
    },
    {
      "id": "dad18f0f-d753-47b7-9393-e30630a394cd",
      "stream_date": "2022-08-27T15:58:41Z",
      "title": "Breaking News: Sports Edition",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/16.jpg",  
      "sport_type": "skiing"
    },
    {
      "id": "18813033-6113-4da8-ad75-2a209da6380f",
      "stream_date": "2024-10-17T20:05:31Z",
      "title": "Live Debate: Game Changers",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/17.jpg",
      "sport_type": "football"
    },
    {
      "id": "449e40c9-1b3a-4a33-931f-5172b5abaf39",
      "stream_date": "2022-08-11T07:26:11Z",
      "title": "Studio Insights: Race Day",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/18.jpg",
      "sport_type": "marathon"
    },
    {
      "id": "386e49d7-3ff1-476a-82b9-b9a61870b3b3",
      "stream_date": "2022-04-30T05:45:24Z",
      "title": "Reporter’s Corner: Match Highlights",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/19.jpg",
      "sport_type": "running"
    },
    {
      "id": "9fbaaba2-b6d0-442e-83fe-529212062a4c",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "On-Air with the Experts",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/20.jpg",
      "sport_type": "soccer"
    },
    {
      "id": "c4253727-d8fe-4b22-b792-e1c79ecaa5f4",
      "title": "In-Studio with Top Analysts",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/24.jpg",
      "sport_type": "boxing"
    },
    {
      "id": "f4a39998-4937-4bb7-82e7-90108a12a702",
      "stream_date": "",
      "title": "Game Day Reporter Spotlight",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/25.jpg",
      "sport_type": "football"
    },
    {
      "id": "39abb98c-922d-4f10-b205-4b95a4d4babb",
      "stream_date": null,
      "title": "Behind the Mic: Sports Talk",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/26.jpg",
      "sport_type": "biking"
    },
    {
      "id": "967d3e67-473d-4351-9c10-3727f55297ee",
      "stream_date": "2022-08-27T15:58:41Z",
      "title": "Breaking News: Sports Edition",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/27.jpg",
      "sport_type": "running"
    },
    {
      "id": "074ac572-c778-4402-bd68-61438f28e10f",
      "stream_date": "2024-10-17T20:05:31Z",
      "title": "Live Debate: Game Changers",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/28.jpg",
      "sport_type": "basketball"
    },
    {
      "id": "c9027268-5ef4-4663-b979-a8ae527373a1",
      "stream_date": "2022-08-11T07:26:11Z",
      "title": "Studio Insights: Race Day",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/29.jpg",
      "sport_type": "baseball"
    },
    {
      "id": "7e1a09a4-1066-43d3-9ee9-0eb89ade5c18",
      "stream_date": "2022-04-30T05:45:24Z",
      "title": "Reporter’s Corner: Match Highlights",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/30.jpg",
      "sport_type": "land hockey"
    },
    {
      "id": "1eeadc2b-4e6f-461a-8f93-825ef54b2f48",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "On-Air with the Experts",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/31.jpg",
      "sport_type": "tennis"
    },
    {
      "id": "144509c0-424f-40e3-ad26-f1ae885e8b79",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "Pre-Game Strategy Breakdown",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/32.jpg",
      "sport_type": "running"
    },
    {
      "id": "93e85028-3a15-44e1-b976-813711a6b938",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "Final Whistle Recap Show",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/33.jpg",
      "sport_type": "biking"
    }
  ]
}