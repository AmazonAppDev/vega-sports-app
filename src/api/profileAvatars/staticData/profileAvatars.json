{
  "profileavatars": [
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/1p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/2p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/3p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/4p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/8p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/9p.jpg"
  ]
}
