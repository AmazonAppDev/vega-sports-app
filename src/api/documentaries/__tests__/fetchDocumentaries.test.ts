import { useQuery } from '@tanstack/react-query';
import { renderHook } from '@testing-library/react-native';

import { AppConfig } from '@AppServices/appConfig';
import {
  useDocumentaries,
  fetchDocumentariesApiCall,
} from '../fetchDocumentaries';
import staticData from '../staticData/documentaries.json';

jest.mock('@tanstack/react-query');

const mockRQ = useQuery as jest.Mock;

describe('fetchDocumentaries', () => {
  beforeAll(() => {
    require('jest-fetch-mock').enableMocks();
  });

  afterEach(() => {
    fetchMock.resetMocks();
  });

  describe('fetchDocumentariesApiCall', () => {
    it('should return parsed documentaries data on success', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.documentaries }),
        {
          status: 200,
        },
      );

      const result = await fetchDocumentariesApiCall();

      expect(result).toMatchSnapshot();
    });

    it('should throw error for 400 status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      fetchMock.mockResponseOnce(JSON.stringify({}), { status: 400 });

      await expect(fetchDocumentariesApiCall()).rejects.toThrow(
        `fetchDocumentariesApiCall(): resources does not exist for endpoint 'documentaries'`,
      );
    });

    it('should throw error for unexpected Api status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      // @ts-expect-error intentionally break TS
      fetchMock.mockResponseOnce(null);

      await expect(fetchDocumentariesApiCall()).rejects.toThrow(
        `Perform a request failed`,
      );
    });

    it('should throw error for other non-200 status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      fetchMock.mockResponseOnce(JSON.stringify({}), { status: 500 });

      await expect(fetchDocumentariesApiCall()).rejects.toThrow(
        `fetchDocumentariesApiCall(): failed to fetch data from endpoint 'documentaries'`,
      );
    });
  });

  describe('useDocumentaries', () => {
    it('should return documentaries data and no error on successful fetch', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.documentaries }),
        {
          status: 200,
        },
      );

      mockRQ.mockReturnValueOnce({
        data: await fetchDocumentariesApiCall(),
        error: null,
        isLoading: false,
      });

      const { result } = renderHook(() => useDocumentaries());

      expect(result.current.data).toMatchSnapshot();
      expect(result.current.error).toBe(null);
      expect(result.current.isLoading).toBe(false);
    });

    it('should return isLoading as true before data loads', () => {
      mockRQ.mockReturnValueOnce({
        data: null,
        error: null,
        isLoading: true,
      });

      const { result } = renderHook(() => useDocumentaries());

      expect(result.current.isLoading).toBe(true);
      expect(result.current.data).toBe(null);
      expect(result.current.error).toBe(null);
    });

    it('should return error when fetch fails', () => {
      const mockError = new Error('Network error');

      mockRQ.mockReturnValueOnce({
        data: null,
        error: mockError,
        isLoading: false,
      });

      const { result } = renderHook(() => useDocumentaries());

      expect(result.current.error).toBe(mockError);
      expect(result.current.data).toBe(null);
      expect(result.current.isLoading).toBe(false);
    });
  });
});
