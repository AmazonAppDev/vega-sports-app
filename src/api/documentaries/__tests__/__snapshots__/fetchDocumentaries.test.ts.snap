// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchDocumentaries fetchDocumentariesApiCall should return parsed documentaries data on success 1`] = `
[
  {
    "itemId": "813f703b-f365-4dae-bbed-60a2af813dae",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/16.jpg",
    "title": "Road to the Championship",
  },
  {
    "itemId": "b1798c7c-cd4b-41a3-85dd-6be0f75fa1e5",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/17.jpg",
    "title": "Beyond the Finish Line",
  },
  {
    "itemId": "a8561206-bdb3-40ef-a3f7-23c920af45fc",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/18.jpg",
    "title": "The Rise of a Superstar",
  },
  {
    "itemId": "9c237dc0-0d91-4aaf-ac96-046ed9b8e495",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/19.jpg",
    "title": "The Science of Winning",
  },
  {
    "itemId": "65c7daeb-9e47-40cc-b61b-b1ea011859b6",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/20.jpg",
    "title": "Chasing Greatness",
  },
  {
    "itemId": "3b7b4569-f755-4d80-885f-41d90eef23ae",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/21.jpg",
    "title": "Legends of the Game",
  },
  {
    "itemId": "165b6aaf-9f93-4022-9910-d3e239e628a4",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/22.jpg",
    "title": "The Journey to Success",
  },
  {
    "itemId": "8e3fe329-cd6a-4e81-a95c-8b40fb8a7f49",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/23.jpg",
    "title": "The Evolution of Football",
  },
  {
    "itemId": "097e7212-200b-4578-9bb9-8fd898203237",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/24.jpg",
    "title": "Behind the Helmet",
  },
  {
    "itemId": "d26ba3e5-7ba8-4e87-897d-df10c16b157e",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/25.jpg",
    "title": "The Golden Era of Basketball",
  },
  {
    "itemId": "0dd81ad5-8914-40c8-b641-b8ff51b67427",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/26.jpg",
    "title": "Racing Through History",
  },
  {
    "itemId": "6aff5300-6eb7-45fc-b810-3da8b2357020",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/27.jpg",
    "title": "The Untold Stories of Sports",
  },
  {
    "itemId": "afcad24f-8e58-4af2-953a-c7074d011beb",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/28.jpg",
    "title": "The Making of a Champion",
  },
  {
    "itemId": "03a73eb9-5543-4087-aecc-50b444e064a7",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/29.jpg",
    "title": "Unstoppable: The Athlete’s Journey",
  },
  {
    "itemId": "5a729bd0-805b-4ca8-b31d-438ae268498a",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/30.jpg",
    "title": "From Grassroots to Glory",
  },
  {
    "itemId": "59b79076-5504-499a-99c5-b8c085d4b53d",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/31.jpg",
    "title": "The Rivalries That Shaped Sports",
  },
  {
    "itemId": "80aa0b2d-05d0-4297-8194-bfbaed28e02b",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/32.jpg",
    "title": "Breaking Barriers in Sports",
  },
  {
    "itemId": "828f58fd-d6e9-4d16-917c-f722461fc268",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/33.jpg",
    "title": "The Art of the Perfect Play",
  },
  {
    "itemId": "220324dd-66eb-4eed-b59f-586fedb31377",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/34.jpg",
    "title": "Dynasties of the Game",
  },
  {
    "itemId": "fc7ee7ed-20a5-49ff-93f0-5fb77828f02d",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/35.jpg",
    "title": "The Heart of a Champion",
  },
  {
    "itemId": "8e43f8b0-fc07-4a1c-b32b-117252fd9e12",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/36.jpg",
    "title": "Unsung Heroes of the Field",
  },
  {
    "itemId": "f9c7114a-23ec-4a9e-8020-a7277e18a985",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/37.jpg",
    "title": "The History of the Olympics",
  },
  {
    "itemId": "7d6e24a6-4bc5-4154-8ada-c9b52b7cf123",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/38.jpg",
    "title": "The Road to Recovery",
  },
  {
    "itemId": "85670ba5-b26f-4796-9b15-f3d55928e29f",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/39.jpg",
    "title": "Game Changers: Innovators in Sports",
  },
  {
    "itemId": "4bec7063-75f9-48fe-b29d-f240cdd95ea3",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/40.jpg",
    "title": "The Spirit of Competition",
  },
  {
    "itemId": "098b2ae8-91a8-4230-9ca4-80c4fc49f706",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/41.jpg",
    "title": "The Rise and Fall of a Legend",
  },
  {
    "itemId": "e8dd0dba-e1f3-446c-83b7-069d005480c6",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/42.jpg",
    "title": "Inside the World of Extreme Sports",
  },
  {
    "itemId": "d6545116-b03c-4263-89cc-79c63cc8b601",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/43.jpg",
    "title": "The Power of Teamwork",
  },
  {
    "itemId": "ff04bf53-bb6f-441a-b9fc-da88bfd55ae3",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/44.jpg",
    "title": "The Psychology of Winning",
  },
  {
    "itemId": "4c562fde-5fb5-4010-ad83-d0c60c3dccb2",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/45.jpg",
    "title": "The Greatest Comebacks in Sports",
  },
  {
    "itemId": "empty-release-date",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/46.jpg",
    "title": "The Business of Sports",
  },
]
`;

exports[`fetchDocumentaries useDocumentaries should return documentaries data and no error on successful fetch 1`] = `
[
  {
    "itemId": "813f703b-f365-4dae-bbed-60a2af813dae",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/16.jpg",
    "title": "Road to the Championship",
  },
  {
    "itemId": "b1798c7c-cd4b-41a3-85dd-6be0f75fa1e5",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/17.jpg",
    "title": "Beyond the Finish Line",
  },
  {
    "itemId": "a8561206-bdb3-40ef-a3f7-23c920af45fc",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/18.jpg",
    "title": "The Rise of a Superstar",
  },
  {
    "itemId": "9c237dc0-0d91-4aaf-ac96-046ed9b8e495",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/19.jpg",
    "title": "The Science of Winning",
  },
  {
    "itemId": "65c7daeb-9e47-40cc-b61b-b1ea011859b6",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/20.jpg",
    "title": "Chasing Greatness",
  },
  {
    "itemId": "3b7b4569-f755-4d80-885f-41d90eef23ae",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/21.jpg",
    "title": "Legends of the Game",
  },
  {
    "itemId": "165b6aaf-9f93-4022-9910-d3e239e628a4",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/22.jpg",
    "title": "The Journey to Success",
  },
  {
    "itemId": "8e3fe329-cd6a-4e81-a95c-8b40fb8a7f49",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/23.jpg",
    "title": "The Evolution of Football",
  },
  {
    "itemId": "097e7212-200b-4578-9bb9-8fd898203237",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/24.jpg",
    "title": "Behind the Helmet",
  },
  {
    "itemId": "d26ba3e5-7ba8-4e87-897d-df10c16b157e",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/25.jpg",
    "title": "The Golden Era of Basketball",
  },
  {
    "itemId": "0dd81ad5-8914-40c8-b641-b8ff51b67427",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/26.jpg",
    "title": "Racing Through History",
  },
  {
    "itemId": "6aff5300-6eb7-45fc-b810-3da8b2357020",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/27.jpg",
    "title": "The Untold Stories of Sports",
  },
  {
    "itemId": "afcad24f-8e58-4af2-953a-c7074d011beb",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/28.jpg",
    "title": "The Making of a Champion",
  },
  {
    "itemId": "03a73eb9-5543-4087-aecc-50b444e064a7",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/29.jpg",
    "title": "Unstoppable: The Athlete’s Journey",
  },
  {
    "itemId": "5a729bd0-805b-4ca8-b31d-438ae268498a",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/30.jpg",
    "title": "From Grassroots to Glory",
  },
  {
    "itemId": "59b79076-5504-499a-99c5-b8c085d4b53d",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/31.jpg",
    "title": "The Rivalries That Shaped Sports",
  },
  {
    "itemId": "80aa0b2d-05d0-4297-8194-bfbaed28e02b",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/32.jpg",
    "title": "Breaking Barriers in Sports",
  },
  {
    "itemId": "828f58fd-d6e9-4d16-917c-f722461fc268",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/33.jpg",
    "title": "The Art of the Perfect Play",
  },
  {
    "itemId": "220324dd-66eb-4eed-b59f-586fedb31377",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/34.jpg",
    "title": "Dynasties of the Game",
  },
  {
    "itemId": "fc7ee7ed-20a5-49ff-93f0-5fb77828f02d",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/35.jpg",
    "title": "The Heart of a Champion",
  },
  {
    "itemId": "8e43f8b0-fc07-4a1c-b32b-117252fd9e12",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/36.jpg",
    "title": "Unsung Heroes of the Field",
  },
  {
    "itemId": "f9c7114a-23ec-4a9e-8020-a7277e18a985",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/37.jpg",
    "title": "The History of the Olympics",
  },
  {
    "itemId": "7d6e24a6-4bc5-4154-8ada-c9b52b7cf123",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/38.jpg",
    "title": "The Road to Recovery",
  },
  {
    "itemId": "85670ba5-b26f-4796-9b15-f3d55928e29f",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/39.jpg",
    "title": "Game Changers: Innovators in Sports",
  },
  {
    "itemId": "4bec7063-75f9-48fe-b29d-f240cdd95ea3",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/40.jpg",
    "title": "The Spirit of Competition",
  },
  {
    "itemId": "098b2ae8-91a8-4230-9ca4-80c4fc49f706",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/41.jpg",
    "title": "The Rise and Fall of a Legend",
  },
  {
    "itemId": "e8dd0dba-e1f3-446c-83b7-069d005480c6",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/42.jpg",
    "title": "Inside the World of Extreme Sports",
  },
  {
    "itemId": "d6545116-b03c-4263-89cc-79c63cc8b601",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/43.jpg",
    "title": "The Power of Teamwork",
  },
  {
    "itemId": "ff04bf53-bb6f-441a-b9fc-da88bfd55ae3",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/44.jpg",
    "title": "The Psychology of Winning",
  },
  {
    "itemId": "4c562fde-5fb5-4010-ad83-d0c60c3dccb2",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/45.jpg",
    "title": "The Greatest Comebacks in Sports",
  },
  {
    "itemId": "empty-release-date",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/300x450/46.jpg",
    "title": "The Business of Sports",
  },
]
`;
